Composer/packages/client/src/pages/botProject/ABSChannels.tsx

```diff
@@ -0,0 +1,674 @@
+        setErrorMessage(subscriptionsResult._response.bodyAsText);
+        return [];
+      }
+      // eslint-disable-next-line no-underscore-dangle
+      return subscriptionsResult._response.parsedBody;
+    } catch (err) {
+      setErrorMessage(JSON.stringify(err, Object.getOwnPropertyNames(err)));
+      return [];
+    }
+  };
+
+  const onSelectProfile = async (_, opt) => {
+    if (opt.key === 'manageProfiles') {
+      navigateTo(`/bot/${projectId}/botProjectsSettings/#addNewPublishProfile`);
+    } else {
+      let newtoken = '';
+      if (isGetTokenFromUser()) {
+        if (isShowAuthDialog(false)) {
+          setShowAuthDialog(true);
+        }
+        newtoken = getTokenFromCache('accessToken');
+      } else {
+        newtoken = await AuthClient.getAccessToken(armScopes);
+      }
+      setToken(newtoken);
+
+      // identify the publishing profile in the list
+      const profile = publishTargets?.find((p) => p.name === opt.key);
+      if (profile) {
+        const config = JSON.parse(profile.configuration);
+        setCurrentResource({
+          resourceName: config.name,
+          resourceGroupName: config.name,
+          subscriptionId: config.subscriptionId,
+        });
+      }
+    }
+  };
+
+  // allow a user to provide a subscription id if one is missing
+  const onChangeSubscription = async (_, opt) => {
+    if (currentResource) {
+      setCurrentResource({
+        ...currentResource,
+        alternateSubscriptionId: opt.key,
+      });
+    }
+  };
+
+  const fetchChannelStatus = async (channelId: string) => {
+    if (currentResource) {
+      try {
+        const url = `https://management.azure.com/subscriptions/${
+          currentResource.subscriptionId || currentResource.alternateSubscriptionId
+        }/resourceGroups/${currentResource?.resourceGroupName}/providers/Microsoft.BotService/botServices/${
+          currentResource?.resourceName
+        }/channels/${channelId}?api-version=2020-06-02`;
+        await httpClient.get(url, { headers: { Authorization: `Bearer ${token}` } });
+        return {
+          enabled: true,
+          loading: false,
+        };
+      } catch (err) {
+        switch (err?.response.data?.error.code) {
+          case 'ResourceNotFound':
+            // this channel has not yet been created, should display as disabled
+            return {
+              enabled: false,
+              loading: false,
+            };
+            break;
+          case 'AuthenticationFailed':
+            // the auth failed for some reason.
+            break;
+          case 'ResourceGroupNotFound':
+            // this resource group is not found - in other words, can't find a channel registration in the expected spot.
+            break;
+          case 'SubscriptionNotFound':
+            // the subscription is not found or invalid
+            break;
+          default:
+            // handle error.
+            break;
```

Comment:
Were you going to add some more error handling here?

Reply:
I was going to consult with design on whether we want to provide custom error messages for these errors I observed.

So yes, probably!?